atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampUtils.java

```diff
@@ -0,0 +1,175 @@
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.cassandra.thrift.Column;
+import org.apache.cassandra.thrift.CqlResult;
+import org.apache.cassandra.thrift.CqlRow;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.table.description.ColumnMetadataDescription;
+import com.palantir.atlasdb.table.description.ColumnValueDescription;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NameMetadataDescription;
+import com.palantir.atlasdb.table.description.NamedColumnDescription;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.util.Pair;
+
+public final class CassandraTimestampUtils {
+    public static final TableMetadata TIMESTAMP_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription("timestamp_name", ValueType.STRING))),
+            new ColumnMetadataDescription(ImmutableList.of(
+                    new NamedColumnDescription(
+                            CassandraTimestampUtils.ROW_AND_COLUMN_NAME,
+                            "current_max_ts",
+                            ColumnValueDescription.forType(ValueType.FIXED_LONG)))),
+            ConflictHandler.IGNORE_ALL);
+
+    private static final long CASSANDRA_TIMESTAMP = -1;
+
+    public static final String ROW_AND_COLUMN_NAME = "ts";
+    private static final String ROW_AND_COLUMN_NAME_HEX_STRING = encodeCassandraHexString(ROW_AND_COLUMN_NAME);
+
+    private static final String APPLIED_COLUMN = "[applied]";
+    private static final String COLUMN_NAME_COLUMN = "column1";
+    private static final String VALUE_COLUMN = "value";
+
+    private static final byte[] SUCCESSFUL_OPERATION = {1};
+
+    public static final String BACKUP_COLUMN_NAME = "oldTs";
+
+    static final byte[] INVALIDATED_VALUE = new byte[1];
```

Comment:
should this have a value in the array? might want to consider using `com.google.protobuf.ByteString` as the field type so that someone doesn't mistakenly mutate the underlying array.